### PR TITLE
Renamed SMC files

### LINKED ISSUE #6171: Rename SMC files
I think we should call sample_smc.py, sampling.py as it's already inside the smc module anyway. smc.py would be better named kernels.py

CC @aloctavodia

### PR DESCRIPTION
**Pull request by IMvision12**
**What is this PR about?**

Renamed SMC Files
Closes : #6171

## Major / Breaking Changes
- Renamed SMC files

### CODE REVIEW

**Comment by ricardoV94** [comment]
Nevermind about the tests, let's keep the old name since it combines the two modules.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -310,7 +310,7 @@ This new version of `Theano-PyMC` comes with an experimental JAX backend which,
...
 - Add sampler stats `process_time_diff`, `perf_counter_diff` and `perf_counter_start`, that record wall and CPU times for each NUTS and HMC sample (see [ #3986](https://github.com/pymc-devs/pymc/pull/3986)).
 - Extend `keep_size` argument handling for `sample_posterior_predictive` and `fast_sample_posterior_predictive`, to work on ArviZ `InferenceData` and xarray `Dataset` input values (see [PR #4006](https://github.com/pymc-devs/pymc/pull/4006) and issue [#4004](https://github.com/pymc-devs/pymc/issues/4004)).
-- SMC-ABC: add the Wasserstein and energy distance functions. Refactor API, the distance, sum_stats and epsilon arguments are now passed `pm.Simulator` instead of `pm.sample_smc`. Add random method to `pm.Simulator`. Add option to save the simulated data. Improved LaTeX representation [#3996](https://github.com/pymc-devs/pymc/pull/3996).
+- SMC-ABC: add the Wasserstein and energy distance functions. Refactor API, the distance, sum_stats and epsilon arguments are now passed `pm.Simulator` instead of `pm.sampling`. Add random method to `pm.Simulator`. Add option to save the simulated data. Improved LaTeX representation [#3996](https://github.com/pymc-devs/pymc/pull/3996).
```
Comment (ricardoV94) [inline RELEASE-NOTES.md]: We don't want to change old release notes

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -411,7 +411,7 @@ Though we had to temporarily remove the `docs/*` folder from the tarball due to
 - Fixed a bug in `Categorical.logp`. In the case of multidimensional `p`'s, the indexing was done wrong leading to incorrectly shaped tensors that consumed `O(n**2)` memory instead of `O(n)`. This fixes issue [#3535](https://github.com/pymc-devs/pymc/issues/3535)
 - Fixed a defect in `OrderedLogistic.__init__` that unnecessarily increased the dimensionality of the underlying `p`. Related to issue issue [#3535](https://github.com/pymc-devs/pymc/issues/3535) but was not the true cause of it.
 - SMC: stabilize covariance matrix [3573](https://github.com/pymc-devs/pymc/pull/3573)
-- SMC: is no longer a step method of `pm.sample` now it should be called using `pm.sample_smc` [3579](https://github.com/pymc-devs/pymc/pull/3579)
+- SMC: is no longer a step method of `pm.sample` now it should be called using `pm.sampling` [3579](https://github.com/pymc-devs/pymc/pull/3579)
```
Comment (ricardoV94) [inline RELEASE-NOTES.md]: Same here

File: pymc/distributions/simulator.py [hunk 3]
```diff
@@ -135,7 +135,7 @@ def simulator_fn(rng, loc, scale, size):
             loc = pm.Normal("loc", 0, 1)
             scale = pm.HalfNormal("scale", 1)
             simulator = pm.Simulator("simulator", simulator_fn, loc, scale, observed=data)
-            idata = pm.sample_smc()
+            idata = pm.sampling()
```
Comment (ricardoV94) [inline pymc/distributions/simulator.py]: Should still be:
```suggestion
            idata = pm.sample_smc()
```

File: pymc/smc/__init__.py [hunk 4]
```diff
@@ -12,4 +12,4 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-from pymc.smc.sample_smc import sample_smc
+from pymc.pymc.smc.sampling import sampling
```
Comment (ricardoV94) [inline pymc/smc/__init__.py]:
```suggestion
from pymc.pymc.smc.sampling import sample_smc
```

File: pymc/smc/sampling.py [hunk 5]
```diff
@@ -33,10 +33,10 @@
 from pymc.backends.base import MultiTrace
 from pymc.model import modelcontext
 from pymc.parallel_sampling import _cpu_count
-from pymc.smc.smc import IMH
+from pymc.pymc.smc.kernels import IMH
 
 
-def sample_smc(
+def sampling(
```
Comment (ricardoV94) [inline pymc/smc/sampling.py]: This function should not be renamed

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
Thanks for opening the PR. We should keep the internal function names unchanged. Change only file names.

**Comment by ricardoV94** [comment]
@IMvision12 Why did you close  #6173 and open this PR?

**Comment by IMvision12** [comment]
No particular reason; did I do something wrong?

File: pymc/smc/__init__.py (hunk 4 above)
Comment (Armavica) [reply pymc/smc/__init__.py]: `pymc.pymc`?

**Comment by Armavica** [comment]
> No particular reason; did I do something wrong?

If you are not happy with a PR that you submitted, or if the tests don't pass, etc., there is no need to create a new PR: the easiest is to push again to the same branch, and the PR will be updated. If you remove or modify some of the commits that are already pushed, you can `git push --force` to replace them. A PR is a request to merge some changes but while it has not been accepted you can always change what you want to merge.

File: pymc/tests/distributions/test_simulator.py [hunk 6]
```diff
@@ -31,7 +31,7 @@
 from pymc import floatX
 from pymc.aesaraf import compile_pymc
 from pymc.initial_point import make_initial_point_fn
-from pymc.smc.smc import IMH
+from pymc.pymc.smc.kernels import IMH
```
Comment (Armavica) [inline pymc/tests/distributions/test_simulator.py]:
```suggestion
from pymc.smc.kernels import IMH
```

File: pymc/smc/sampling.py [hunk 7]
```diff
@@ -33,7 +33,7 @@
 from pymc.backends.base import MultiTrace
 from pymc.model import modelcontext
 from pymc.parallel_sampling import _cpu_count
-from pymc.smc.smc import IMH
+from pymc.pymc.smc.kernels import IMH
```
Comment (Armavica) [inline pymc/smc/sampling.py]:
```suggestion
from pymc.smc.kernels import IMH
```

**Comment by IMvision12** [comment]
Thanks for Information

File: pymc/tests/smc/test_smc.py [hunk 8]
```diff
@@ -24,7 +24,7 @@
 
 from pymc.aesaraf import floatX
 from pymc.backends.base import MultiTrace
-from pymc.smc.smc import IMH
+from pymc.pymc.smc.kernels import IMH
```
Comment (Armavica) [inline pymc/tests/smc/test_smc.py]:
```suggestion
from pymc.smc.kernels import IMH
```

**Review by Armavica — CHANGES_REQUESTED** [review]

File: pymc/distributions/simulator.py [hunk 9]
```diff
@@ -66,7 +66,7 @@ def sum_stat(cls, *args, **kwargs):
 class Simulator(Distribution):
     r"""
     Simulator distribution, used for Approximate Bayesian Inference (ABC)
-    with Sequential Monte Carlo (SMC) sampling via :func:`~pymc.sample_smc`.
+    with Sequential Monte Carlo (SMC) sampling via :func:`~pymc.sampling`.
```
Comment (Armavica) [inline pymc/distributions/simulator.py]: This refers to the function, not to the module, so it shouldn't be changed.

File: pymc/smc/sampling.py [hunk 10]
```diff
@@ -142,7 +142,7 @@ def sample_smc(
 
     if isinstance(kernel, str) and kernel.lower() in ("abc", "metropolis"):
         warnings.warn(
-            f'The kernel string argument "{kernel}" in sample_smc has been deprecated. '
+            f'The kernel string argument "{kernel}" in sampling has been deprecated. '
```
Comment (Armavica) [inline pymc/smc/sampling.py]: This also shouldn't be changed.

File: pymc/tests/distributions/test_simulator.py [hunk 11]
```diff
@@ -78,7 +78,7 @@ def test_one_gaussian(self):
         assert self.count_rvs(self.SMABC_test.logp()) == 1
 
         with self.SMABC_test:
-            trace = pm.sample_smc(draws=1000, chains=1, return_inferencedata=False)
+            trace = pm.sampling(draws=1000, chains=1, return_inferencedata=False)
```
Comment (Armavica) [inline pymc/tests/distributions/test_simulator.py]: This is also the function, and shouldn't be changed (in all this file).

File: pymc/tests/smc/test_smc.py [hunk 12]
```diff
@@ -75,7 +75,7 @@ def two_gaussians(x):
     def test_sample(self):
         initial_rng_state = np.random.get_state()
         with self.SMC_test:
-            mtrace = pm.sample_smc(draws=self.samples, return_inferencedata=False)
+            mtrace = pm.sampling(draws=self.samples, return_inferencedata=False)
```
Comment (Armavica) [inline pymc/tests/smc/test_smc.py]: Same thing in this file.

File: docs/source/api/smc.rst [hunk 13]
```diff
@@ -6,13 +6,13 @@ Sequential Monte Carlo
 .. autosummary::
    :toctree: generated/
 
-    sample_smc
+    sampling
```
Comment (Armavica) [inline docs/source/api/smc.rst]: I think that this also shouldn't be changed.

**Review by Armavica — CHANGES_REQUESTED** [review]
Following up on @ricardoV94's review, I pointed out the places that should not be modified.

**Comment by IMvision12** [comment]
@Armavica Thanks for feedback, I have updated the files

File: pymc/tests/distributions/test_simulator.py (hunk 11 above)
Comment (Armavica) [reply pymc/tests/distributions/test_simulator.py]: Sorry for not being clear but this comment applies to the whole file. You should change back all of the places where you replaced `sample_smc` by `sampling` in this file. As @ricardoV94 was saying, it is just the file name that needs to be changed, not the name of the function.

File: pymc/tests/smc/test_smc.py (hunk 12 above)
Comment (Armavica) [reply pymc/tests/smc/test_smc.py]: Same thing in this file.

File: pymc/tests/smc/test_smc.py [hunk 14]
```diff
@@ -235,13 +235,13 @@ def test_deprecated_abc_args(self):
         with self.fast_model:
             with pytest.warns(
                 FutureWarning,
-                match='The kernel string argument "ABC" in sample_smc has been deprecated',
+                match='The kernel string argument "ABC" in sampling has been deprecated',
```
Comment (Armavica) [inline pymc/tests/smc/test_smc.py]:
```suggestion
                match='The kernel string argument "ABC" in sample_smc has been deprecated',
```

File: pymc/tests/smc/test_smc.py [hunk 15]
```diff
@@ -235,13 +235,13 @@ def test_deprecated_abc_args(self):
...
             ):
                 pm.sample_smc(draws=10, chains=1, kernel="ABC")
 
             with pytest.warns(
                 FutureWarning,
-                match='The kernel string argument "Metropolis" in sample_smc has been deprecated',
+                match='The kernel string argument "Metropolis" in sampling has been deprecated',
```
Comment (Armavica) [inline pymc/tests/smc/test_smc.py]:
```suggestion
                match='The kernel string argument "Metropolis" in sample_smc has been deprecated',
```

**Review by Armavica — CHANGES_REQUESTED** [review]
There are still two places in `test_smc.py` that have been modified but shouldn't be. There are also a few places in this file that have not been modified but should be, you will see them if you run the tests with `pytest pymc/tests/smc`. After that, I think that it should be good.

**Comment by IMvision12** [comment]
@Armavica I modified the file, but the tests are still failing.

**Comment by Armavica** [comment]
Some of these tests fail because there are still a few places in `test_smc.py` that should be updated as a consequence of the file name change. You can see the output of the relevant tests on your computer if you run `pytest pymc/tests/smc`, or also here: https://github.com/pymc-devs/pymc/actions/runs/3188204616/jobs/5201000626 Please let me know if you need help fixing these issues.

Some others fail because now `pymc.sampling` is colliding with `pymc.smc.sampling`, because of https://github.com/pymc-devs/pymc/blob/a135f53b51ea4e6c2b32ec37d95efec015a7a6c1/pymc/__init__.py#L71
@ricardoV94 what should we do here?

**Comment by ricardoV94** [comment]
> Some of these tests fail because there are still a few places in `test_smc.py` that should be updated as a consequence of the file name change. You can see the output of the relevant tests on your computer if you run `pytest pymc/tests/smc`, or also here: https://github.com/pymc-devs/pymc/actions/runs/3188204616/jobs/5201000626 Please let me know if you need help fixing these issues.
> 
> Some others fail because now `pymc.sampling` is colliding with `pymc.smc.sampling`, because of
> 
> https://github.com/pymc-devs/pymc/blob/a135f53b51ea4e6c2b32ec37d95efec015a7a6c1/pymc/__init__.py#L71
> 
> @ricardoV94 what should we do here?

I pushed some changes that specify `__all__` so that not everything is imported at the root level

**Comment by ricardoV94** [comment]
@IMvision12 I squashed the commits and force-pushed do solve the conflicts. Be careful to get the upstream changes before you try to do further changes

File: pymc/smc/__init__.py [hunk 16]
```diff
@@ -12,4 +12,7 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-from pymc.smc.sample_smc import sample_smc
+from pymc.smc.kernels import IMH, MH
+from pymc.smc.sampling import sample_smc
+
+__all__ = ("sample_smc",)
```
Comment (michaelosthege) [inline pymc/smc/__init__.py:18]: @IMvision12 you didn't include `IMH` and `MH` in `__all__`?

**Review by michaelosthege — APPROVED** [review]
One small comment, otherwise this appears to be ready to merge?

File: pymc/smc/__init__.py (hunk 16 above)
Comment (ricardoV94) [reply pymc/smc/__init__.py:18]: They weren't available at the root level before, just via pymc.smc. I think it was fine like that

Comment (Armavica) [inline pymc/smc/__init__.py:18]: nit: Isn't `__all__` usually a list? In PyMC it is a list in 42 places and a tuple in only 2.

**Review by Armavica — APPROVED** [review]

File: pymc/smc/__init__.py (hunk 16 above)
Comment (ricardoV94) [reply pymc/smc/__init__.py:18]: I'm team tuples

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**